packages/cloudflare/src/utils/image-config.ts

```diff
@@ -0,0 +1,37 @@
+import type { AstroIntegrationLogger } from 'astro';
+import { passthroughImageService, sharpImageService } from 'astro/config';
+
+export function prepareImageConfig(
+	service: string,
+	// FIXME: type
+	config: any,
+	// FIXME: type
+	command: any,
+	logger: AstroIntegrationLogger
+) {
+	switch (service) {
+		case 'passthrough':
+			return { ...config.image, service: passthroughImageService() };
+
+		case 'external':
+			return {
+				...config.image,
+				service:
+					command === 'dev'
+						? sharpImageService()
+						: { entrypoint: '@astrojs/cloudflare/image-service' },
+			};
+
+		default:
+			if (
+				config.service.entrypoint === 'astro/assets/services/sharp' ||
```

Comment:
Is `config` the astro config or specifically the image config?

Reply:
See https://github.com/withastro/adapters/pull/34/files#r1387090220

Reply:
Yeah, that's why I bring it up. It is being used both as the image config `config.service`, and as the main config `config.image`.

Reply:
Good catch!